packages/babel-plugin-proposal-class-properties/test/fixtures/general/computed/expected.js

```diff
@@ -6,7 +6,7 @@ var four = 4;
 
 var MyClass =
 /*#__PURE__*/
```

Comment:
Isn't this iife pure only if computed properties are pure?

Reply:
Yes, you are right. They can also be impure in other cases, such as impure property initializers, or even super class being an impure expression.

I've wanted to keep marking those class IIFEs as pure, because it is what transform-classes do at the moment in all cases (and classes without properties can be impure too). That's quite my fault, adding those annotations was one of my 1st PRs to babel.

I'd like to revisit (as another PR) annotating classes as pure as a whole and not just in case of proposal-class-properties plugin. While checking if class is pure should be quite easy, I'm not sure what we should do about subclasses - we won't be able to tell if a super class is pure or not.